lib/agent.js

```diff
@@ -40,6 +41,8 @@ function Agent () {
   this._uncaughtExceptionListener = null
 
   this._config()
+
+  this.lambda = lambda(this)
```

Comment:
The intention is for it to be pulled out to another module, so the _eventual_ usage for it would just be:

```js
const agent = require('elastic-apm-node').start()
const lambda = require('some-module-name')(agent)

exports.handler = lambda(async payload => {
  return { hello: 'world' }
})
```

Due to this design of returning the same function as the original, bound to the given agent, I chose to go this route of setting it in the constructor. Alternatively, I could do something like:

```js
this._lambda = lambda(this)

// ...later...

Agent.prototype.lambda = function lambda (...args) {
  return this._lambda(...args)
}
```

That seems like a needless extra step though. 🤔 

What do you think?

Reply:
> That seems like a needless extra step though. 🤔

I thought the same thing. It was just because we currently do that for all the methods on the `Instrumentation` object. But 5 wrongs doesn't make a right of course 😂 

If `Agent` wasn't a singleton it would be a different story of course, as there would be benefit of of having it on the prototype, but I don't think we're missing out of any benefits in this case right?

Reply:
I'm unclear what the decision is then--do we _want_ a prototype method, or is the current code fine?

Reply:
I think this code is fine. We should just look into aligning how we do this later